plugin/plugin.go

```diff
@@ -21,47 +18,29 @@ import (
 	"github.com/weaveworks/weave/plugin/skel"
 )
 
-var version = "unreleased"
-
 var Log = common.Log
 
-func main() {
-	var (
-		justVersion      bool
-		address          string
-		meshAddress      string
-		logLevel         string
-		noMulticastRoute bool
-	)
-
-	flag.BoolVar(&justVersion, "version", false, "print version and exit")
-	flag.StringVar(&logLevel, "log-level", "info", "logging level (debug, info, warning, error)")
-	flag.StringVar(&address, "socket", "/run/docker/plugins/weave.sock", "socket on which to listen")
-	flag.StringVar(&meshAddress, "meshsocket", "/run/docker/plugins/weavemesh.sock", "socket on which to listen in mesh mode")
-	flag.BoolVar(&noMulticastRoute, "no-multicast-route", false, "deprecated (this is now the default)")
-
-	flag.Parse()
-
-	if justVersion {
-		fmt.Printf("weave plugin %s\n", version)
-		os.Exit(0)
-	}
-
-	common.SetLogLevel(logLevel)
+func Start(weaveAPIAddr string, dockerAPIAddr string, address string, meshAddress string) {
+	weave := weaveapi.NewClient(weaveAPIAddr, Log)
```

Comment:
It is in my TODO. For this PR I didn't want to introduce any other significant changes to prevent from possible bugs.